packages/server/modules/webhooks/services/webhooks.ts

```diff
@@ -113,47 +114,48 @@ export const dispatchStreamEventFactory =
     createWebhookEvent: CreateWebhookEvent
     getUser: typeof getUserFn
   }) =>
-  async (
-    {
-      streamId,
-      event,
-      eventPayload
-    }: {
-      streamId: string
-      event: string
-      eventPayload: {
-        server: { id?: number; canonicalUrl?: string }
-        streamId?: string
-        stream?: StreamWithOptionalRole
-        userId?: string
-        user: Partial<UserWithOptionalRole> | null
-        webhook: Webhook
-      }
-    },
-    { trx }: { trx?: Knex.Transaction } = {}
-  ) => {
+  async ({
+    streamId,
+    event,
+    eventPayload
+  }: {
+    streamId: string
+    event: string
+    eventPayload: {
+      streamId?: string
+      stream?: StreamWithOptionalRole
+      userId?: string | null
+      user?: Partial<UserWithOptionalRole> | null
+    }
+  }) => {
+    const payload: typeof eventPayload & {
+      server: Partial<Omit<ServerInfo, 'secret'>>
+    } = {
+      ...eventPayload,
+      server: { ...(await getServerInfo()), canonicalUrl: process.env.CANONICAL_URL }
+    }
     // Add server info
-    eventPayload.server = await getServerInfo()
-    eventPayload.server.canonicalUrl = process.env.CANONICAL_URL
-    delete eventPayload.server.id
+    payload.server = await getServerInfo()
+    payload.server.canonicalUrl = process.env.CANONICAL_URL
+    delete payload.server.id
 
     // Add stream info
-    if (eventPayload.streamId) {
-      eventPayload.stream = await getStream(
+    if (payload.streamId) {
+      payload.stream = await getStream(
         {
-          streamId: eventPayload.streamId,
-          userId: eventPayload.userId
+          streamId: payload.streamId,
+          userId: payload.userId ?? undefined
         },
-        { trx }
+        { trx: db.isTransaction ? await db.transaction() : undefined }
```

Comment:
I used this trick to avoid exposing the `trx` object in the function since we will eliminate it during the refactor

Reply:
nice, hadn't thought about the fact that you can pass in a transaction as the `db` dependency